### PR TITLE
Adding functionality to allow custom brokers to be registered in Herman generically

### PR DESCRIPTION
**Pull request by binaryblake**
Basically the idea here is that you can "register" a broker in your properties.yml file, which get picked up during deployment at the appropriate phase by Herman. We have some internal, LM specific use cases for this. 

Here is an example of a custom broker defined in properties.yml:

```
customBrokers:
  herman-rds-broker-papi:
    description: >-
      This plugin is meant for internal use at Liberty Mutual. It provisions PAPI compliant RDS instances.
      Depending on your configuration, it may take 10 minutes or longer to provision or update an instance.
    phase: "PREPUSH"
    variablesToPass:
      "bamboo.secret.papi-index": "papiIndex"
      "bamboo.secret.vault-token.token": "papiVaultToken"
      "bamboo.forge.deployment.guid": "artifactDeploymentGuid"
      "bamboo.deploy.environment": "appEnv"
    defaults:
      vpcId: vpc-06b8e561
      subnetIds:
        - subnet-1cf71b47
        - subnet-5c07d915
        - subnet-0e331aad3f01c56c2
      sbu: USCM
```

Brokers can be registered PREPUSH or POSTPUSH, and it would be pretty easy to extend that if needed. VariablesToPass is optional and allows passing of pipline environment variables that may be needed for the custom broker.

"Defaults" is where you can defined default values that can be overriden in the app-specific props, but don't need to be defined.

Here is an example of an app using that custom broker definition in template.yml:

```
customBrokers:
  - name: herman-rds-broker-papi
    properties:
      instance:
        DBInstanceIdentifier: testingdb
        Engine: mysql
        EngineVersion: 5.7
```

Let me know what you guys think... sorry for the huge PR! Open to any thoughts/suggestions

### CODE REVIEW

**Comment by fritzdj** [comment]
I like where you are going with this @binaryblake. Wondering if we should take the next step and make this standard for all brokers?

File: src/test/resources/sampleTemplates/properties.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+company: lm
+sbu: USCM
+org: Products
+engine: LMBDT
+rdsCredentialBrokerImage: "repo.forge.lmig.com/docker/grm/ah/ecs/rds-credential-broker:BUILD6-Rb63a8812e36fc465e855ca5066e753bf9fef3c0f"
```
Comment (fritzdj) [inline src/test/resources/sampleTemplates/properties.yml]: Wondering if we should keep LM specifics like a bunch of values in this file out of the GH repo

Comment (binaryblake) [reply src/test/resources/sampleTemplates/properties.yml]: Yep, absolutely - I'll make this generic instead of having LM values.

**Comment by binaryblake** [comment]
Yeah I think it could make sense @fritzdj . I'll give it a go with a single custom broker and maybe we can see what makes sense to try and externalize next? I'm sure there will need to be some tweaks to it, but I think it would help with decoupling and making things a little more flexible around the core framework.

**Pull request merged**